File: code_examples/F2PY/f2py_examples/moddata.py
import moddata

print 'mod.__doc__'
print moddata.mod.__doc__

moddata.mod.i = 5
moddata.mod.x[:2] = [1,2]
moddata.mod.a = [[1,2,3],[4,5,6]]

print 'exec foo()'
moddata.mod.foo()

print '\na='
print moddata.mod.a

print 'exec bar()'
moddata.mod.bar()

print 'set b'
moddata.mod.b = [[7,8,9], [18,9,2]]
moddata.mod.bar()

print 'set b with numpy'
import numpy as np
moddata.mod.b = np.random.random((3,4))
moddata.mod.bar()

print 'set b as None'
moddata.mod.b = None
moddata.mod.bar()

File: code_examples/F2PY/f2py_examples/moddata.f90
MODULE mod
  INTEGER i
  INTEGER :: x(4)
  REAL, DIMENSION(2,3) :: a
  REAL, ALLOCATABLE, DIMENSION(:,:) :: b

CONTAINS
  SUBROUTINE foo
    INTEGER k
    PRINT*, "i=", i
    PRINT*, "x=[", X, "]"
    PRINT*, "a=["
    PRINT*, "[", a(1,1), ",", a(1,2), ",", a(1,3), "]"
    PRINT*, "[", a(2,1), ",", a(2,2), ",", a(2,3), "]"
    PRINT*, "]"
    PRINT*, "Setting a(1,2)+3"
    a(1,2) = a(1,2) + 3
  END SUBROUTINE foo

  SUBROUTINE bar
    INTEGER k
    IF( ALLOCATED(b) ) THEN
      PRINT*, "b=["
      DO k = 1,size(b,1)
        PRINT*, B(k, 1:size(b,2))
      ENDDO
      PRINT*, "]"
    ELSE
      PRINT*, "b is not allocated"
    ENDIF
  END SUBROUTINE bar
END MODULE mod
